src/dft/backends/mklcpu/mklcpu_helpers.hpp

```diff
@@ -0,0 +1,221 @@
+                                    "Invalid config value for complex storage.");
+        return 0;
+    }
+}
+
+template <>
+inline constexpr int to_mklcpu<dft::detail::config_param::REAL_STORAGE>(
+    dft::detail::config_value value) {
+    if (value == dft::detail::config_value::REAL_REAL) {
+        return DFTI_REAL_REAL;
+    }
+    else {
+        throw mkl::invalid_argument("dft", "MKLCPU descriptor set_value()",
+                                    "Invalid config value for real storage.");
+        return 0;
+    }
+}
+template <>
+inline constexpr int to_mklcpu<dft::detail::config_param::CONJUGATE_EVEN_STORAGE>(
+    dft::detail::config_value value) {
+    if (value == dft::detail::config_value::COMPLEX_COMPLEX) {
+        return DFTI_COMPLEX_COMPLEX;
+    }
+    else {
+        throw mkl::invalid_argument("dft", "MKLCPU descriptor set_value()",
+                                    "Invalid config value for conjugate even storage.");
+        return 0;
+    }
+}
+
+template <>
+inline constexpr int to_mklcpu<dft::detail::config_param::PLACEMENT>(
+    dft::detail::config_value value) {
+    if (value == dft::detail::config_value::INPLACE) {
+        return DFTI_INPLACE;
+    }
+    else if (value == dft::detail::config_value::NOT_INPLACE) {
+        return DFTI_NOT_INPLACE;
+    }
+    else {
+        throw mkl::invalid_argument("dft", "MKLCPU descriptor set_value()",
+                                    "Invalid config value for inplace.");
+        return 0;
+    }
+}
+
+template <>
+inline constexpr int to_mklcpu<dft::detail::config_param::PACKED_FORMAT>(
+    dft::detail::config_value value) {
+    if (value == dft::detail::config_value::CCE_FORMAT) {
+        return DFTI_CCE_FORMAT;
+    }
+    else {
+        throw mkl::invalid_argument("dft", "MKLCPU descriptor set_value()",
+                                    "Invalid config value for packed format.");
+        return 0;
+    }
+}
+
+enum DIR { fwd, bwd };
+
+template <dft::detail::precision prec, dft::detail::domain dom>
+class commit_derived_impl : public dft::detail::commit_impl<prec, dom> {
+private:
+    static constexpr DFTI_CONFIG_VALUE mklcpu_prec = to_mklcpu(prec);
+    static constexpr DFTI_CONFIG_VALUE mklcpu_dom = to_mklcpu(dom);
+    using mklcpu_desc_t = DFTI_DESCRIPTOR_HANDLE;
+
+public:
+    commit_derived_impl(sycl::queue queue, const dft::detail::dft_values<prec, dom>& config_values);
+
+    virtual void commit(const dft::detail::dft_values<prec, dom>& config_values) override;
+
+    void* get_handle() noexcept override;
+
+    virtual ~commit_derived_impl() override;
+
+    virtual sycl::buffer<std::vector<mklcpu_desc_t>, 1> get_handle_buffer() noexcept;
+
+private:
+    std::vector<mklcpu_desc_t> bidirection_handle{ nullptr, nullptr };
+    sycl::buffer<std::vector<mklcpu_desc_t>, 1> bidirection_buffer{ &bidirection_handle,
+                                                                    sycl::range<1>{ 1 } };
```

Comment:
```suggestion
    // bidirectional_handle[0] is the forward handle, bidirectional_handle[1] is the backward handle
    std::array<mklcpu_desc_t,2> bidirectional_handle{ nullptr, nullptr };
    sycl::buffer<mklcpu_desc_t, 1> bidirection_buffer{ &bidirection_handle,
                                                                    sycl::range<1>{ 2 } };
```

Reply:
Just to clarify, we don't need a vector here because the size is always 2. This avoids an unneeded allocation and conveys the intent that we have a fixed known size.

Reply:
The buffer is already a list, the specialisation should be `sycl::buffer<mklcpu_desc_t,1>` as I suggested.